### DISCUSSION TITLE
unpoly.com website is broken on Safari – let me help!

### DISCUSSION THREAD

**Discussion by codepilotsf** [unpoly/unpoly#685] [General]
Like most web devs, I use Chrome for work. But I use Safari as my daily browser on my Mac and my iPhone because it's nicely integrated with the OS and with the Passwords (Keychain) app. On both MacOS and iOS, the unpoly.com website is in pretty rough shape. Clicking through the docs leaves a huge blue focus ring around the newly loaded content until you refresh. On iOS Safari, there are lots of weird side-scrolling issues which actually make it tough to read the docs. I'm pointing this out as someone who checked out Unpoly a year or two ago and immediately judged it to be not a serious project because of the state of the website. After hearing Hennen on the HX-Pod podcast, Unpoly strikes me as a _very_ serious project worthy of more attention! But the website doesn't do it justice – especially for Safari users where the website is straight up broken which doesn't make a great first impression. Let me know how I can help fix it up!

**Comment by triskweline** [comment]
I think I fixed the blue focus outline on Safari, and the horizontal scroll bars on iOS.

If you see anything else that needs addressing, the entire site open-sourced under [unpoly/unpoly-site](https://github.com/unpoly/unpoly-site).

**Comment by codepilotsf** [comment]
Oh nice work! Just did a quick spot-check on Safari MacOS and iOS and they both look great. Curious, was the blue outline thing something specific to Unpoly and Safari? Something folks who stumble across this thread might like to know how to fix down the road?

**Reply by hpvd** [reply]
see new release https://github.com/unpoly/unpoly/discussions/686

**Comment by triskweline** [comment]
> Curious, was the blue outline thing something specific to Unpoly and Safari?

- Unpoly [often focuses new content](https://unpoly.com/focus#automatic-focus-logic) for accessibility reasons.
- Browsers draw focus outlines around elements focused by JavaScript.
- Unpoly [hides these focus outlines for mouse users](https://unpoly.com/focus-visibility).
- Hiding outlines turned out to be non-trivial. 3.9.4 fixed an edge case in sites that have transforms on the focus outlines, but ended up breaking it in Safari instead. Fixed in 3.9.5.
